.github/report-flaky-tests/index.js

```diff
@@ -0,0 +1,249 @@
+			} );
+		}
+	}
+} )().catch( ( err ) => {
+	core.error( err );
+} );
+
+async function downloadReportFromArtifact( octokit, artifactName ) {
+	const {
+		data: { artifacts },
+	} = await octokit.rest.actions.listWorkflowRunArtifacts( {
+		...github.context.repo,
+		run_id: github.context.payload.workflow_run.id,
+	} );
+
+	const matchArtifact = artifacts.find(
+		( artifact ) => artifact.name == artifactName
+	);
+
+	if ( ! matchArtifact ) {
+		return;
+	}
+
+	const download = await octokit.rest.actions.downloadArtifact( {
+		...github.context.repo,
+		artifact_id: matchArtifact.id,
+		archive_format: 'zip',
+	} );
+
+	const {
+		files: [ file ],
+	} = await unzipper.Open.buffer( Buffer.from( download.data ) );
+	const buffer = await file.buffer();
+
+	return JSON.parse( buffer.toString() );
+}
+
+function getIssueTitle( testTitle ) {
+	return `[Flaky Test] ${ testTitle }`;
+}
+
+function getTestPathAndTitle( testKey ) {
+	const separatorIndex = testKey.indexOf( ':' );
+	return [
+		testKey.slice( 0, separatorIndex ),
+		testKey.slice( separatorIndex + 1 ),
+	];
+}
+
+function renderIssueBody( { meta, testTitle, testPath, testResults } ) {
+	return (
+		renderIssueDescription( { meta, testTitle, testPath } ) +
+		renderTestResults( { testPath, testResults } )
+	);
+}
+
+function renderIssueDescription( { meta, testTitle, testPath } ) {
+	return `${ metaData.render( meta ) }
+**Flaky test detected. This is an auto-generated issue by GitHub Actions. Please do NOT edit this manually.**
+
+## Test title
+${ testTitle }
+
+## Test path
+\`${ testPath }\`
+
+## Flaky rate (__estimated__)
+\`${ meta.failedTimes } / ${ meta.totalCommits + meta.failedTimes }\`
+
+## Errors
+`;
+}
+
+function renderTestResults( { testPath, testResults } ) {
+	return `${ TEST_RESULTS_LIST.open }
+${ renderTestErrorMessage( { testPath, testResults } ) }
+${ TEST_RESULTS_LIST.close }
+`;
+}
+
+function renderTestErrorMessage( { testPath, testResults } ) {
+	const date = new Date().toISOString();
+	return `${ TEST_RESULT.open }<details>
+<summary>
+	<time datetime="${ date }"><code>[${ date }]</code></time>
+		Test passed after ${ testResults.length } failed ${
+		testResults.length === 0 ? 'attempt' : 'attempts'
+	} on <a href="${ getRunURL() }"><code>${ getHeadBranch() }</code></a>.
+</summary>
+
+\`\`\`
+${ stripAnsi(
+	formatResultsErrors(
+		testResults,
+		{ rootDir: path.join( process.cwd(), 'packages/e2e-tests' ) },
+		{},
+		testPath
+	)
+) }
+\`\`\`
+</details>${ TEST_RESULT.close }`;
+}
+
+function getHeadBranch() {
+	return github.context.payload.workflow_run.head_branch;
+}
+
+function getRunURL() {
+	return github.context.payload.workflow_run.html_url;
+}
+
+function stripAnsi( string ) {
+	return string.replace(
+		new RegExp(
+			[
+				'[\\u001B\\u009B][[\\]()#;?]*(?:(?:(?:[a-zA-Z\\d]*(?:;[-a-zA-Z\\d\\/#&.:=?%@~_]*)*)?\\u0007)',
+				'(?:(?:\\d{1,4}(?:;\\d{0,4})*)?[\\dA-PR-TZcf-ntqry=><~]))',
+			].join( '|' ),
+			'g'
+		)
```

Comment:
![yes](https://media.giphy.com/media/KDVswimTNahWzcd7sV/giphy.gif?cid=ecf05e47vtdzqwilewt3aw6g6i5yhlnpt0asd80oexc04tno&rid=giphy.gif&ct=g)

Reply:
Oh, this is actually copied from [`strip-ansi`](https://github.com/chalk/strip-ansi) but it requires ES modules, which means we need a build step. I don't want to do that for now but I'll switch to using it eventually.

Reply:
Could we use one of these packages which don't seem to require ES modules?

https://www.npmjs.com/package/strip-ansi-control-characters
https://www.npmjs.com/package/strip-color

If that's not feasible, then we need to at least add a comment above this function that contains the copyright notice.

https://github.com/chalk/strip-ansi/blob/main/license